projects/storefrontlib/src/lib/my-account/order/order-details/order-detail-items/order-detail-items.component.ts

```diff
@@ -1,22 +1,42 @@
 import { Component, OnInit } from '@angular/core';
 import { Observable } from 'rxjs';
 
-import { Order, Consignment, OrderEntry } from '@spartacus/core';
+import {
+  Order,
+  Consignment,
+  OrderEntry,
+  ConsignmentTracking,
+  UserService,
+} from '@spartacus/core';
+
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
 
 import { OrderDetailsService } from '../order-details.service';
+import { TrackingEventsComponent } from './consignment-tracking/tracking-events.component';
+import { tap } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-order-details-items',
   templateUrl: './order-detail-items.component.html',
   styleUrls: ['./order-detail-items.component.scss'],
 })
 export class OrderDetailItemsComponent implements OnInit {
-  constructor(private orderDetailsService: OrderDetailsService) {}
+  constructor(
+    private orderDetailsService: OrderDetailsService,
+    private userService: UserService,
+    private ngbModal: NgbModal
+  ) {}
 
   order$: Observable<Order>;
+  consignmentTracking$: Observable<ConsignmentTracking>;
+  orderCode: string;
+  dialog: any;
```

Comment:
You could specify type here for typesafety

Reply:
but `ngbModal.open(....).componentInstance` is really `any` type.